client/src/main/java/client/ClientControl.java

```diff
@@ -0,0 +1,93 @@
+package client;
+
+import client.sprite.PlayerCommandsTypeCoder;
+import engine.player.command.PlayerCommandType;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.util.Collection;
+import java.util.HashSet;
+
+class ClientControl extends Thread {
+
+    private final static int MS_PER_UPDATE = 30;
+
+    private Socket clientSocket;
+    private PrintWriter out;
+    private BufferedReader in;
+
+    private final PlayerInput playerInput;
+
+    private volatile boolean running;
+
+    ClientControl(PlayerInput playerInput) {
+        this.playerInput = playerInput;
+    }
+
+    @Override
+    public void run() {
+
+        this.running = true;
+        double lag = 0.0;
+        double previous = getCurrentTime();
+        while (isRunning()) {
+            double current = getCurrentTime();
+            double elapsed = current - previous;
+            previous = current;
+            lag += elapsed;
+
+            while (lag >= MS_PER_UPDATE) {
+                String message = formMessage();
+
+                try {
+                    sendMessage(message);
+
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                lag -= MS_PER_UPDATE;
+            }
+        }
+
+        try {
+            stopConnection();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    void startConnection(String ip, int port) throws IOException {
```

Comment:
ClientControl is responsible for processing player commands inputs. Create separate ClientConnection class

Reply:
IP is protocol. 

String ip -> String host or String ip -> String address

Reply:
argument fields are always DEFAULT_HOST, DEFAULT_PORT. Make DEFAULT_HOST, DEFAULT_PORT as private static final fields in ClientConnection class